### PR TITLE
RzCmd: add support for dynamically generated details sections

### LINKED ISSUE #2359: Generate help of a command at runtime
**Is your feature request related to a problem? Please describe.**
Some commands like `woE` and `woD` accept as `algo` argument one of the supported crypto algorithms, which are dynamic based on which plugins are loaded in Rizin. I would like to have the algorithms listed as part of `woE?`, in a `detail` section of the command help, however right now the detail part is just static, defined in the YAML files.

**Describe the solution you'd like**
In the YAML file:
* if `details` contains a string, than that's the name of a C handler that returns an array of `RzCmdDescDetail`.
In the C structures:
* add a `details_cb` in `RzCmdDescHelp`.

The details_cb should be something like:
```
RzCmdDescDetail *detail_cb(RzCore *core)
```

**Describe alternatives you've considered**
New command just for this, but it isn't ideal IMO.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

So far all the help sections were statically generated from the YAML
files, however sometimes you may want to list things in the help
messages that depend on what is loaded in Rizin (e.g. plugins, etc.).
This patch adds support for a details_cb that provides a dynamically
created RzCmdDescDetail array.

This will be useful also for `woD`/`woE` and probably for other things where adding a new help section makes sense.
image

**Test plan**

Run `#!?`

**Closing issues**

Closes #2359

### CODE REVIEW

**Comment by ret2libc** [comment]
I'm not 100% happy with the way I had to free `RzCmdDescDetail`, due to it being all `const`, since originally all the structures were static. Do you have less hackish ways to do this? @thestr4ng3r @wargio ? See all the various casts from/to `(const char *)`/`(char *)` or `(const RzCmdDescDetailEntry *)`.

File: librz/core/cmd/cmd_api.c [hunk 1]
```diff
@@ -2619,3 +2647,32 @@ RZ_API void rz_cmd_state_output_print(RzCmdStateOutput *state) {
...
+ *
+ * Usually command handlers do not need to free the \p details sections because
+ * they are const, but due to \p details_cb those sections could be dynamically
+ * generated. In that case all the data within the details should be dynamically
+ * allocated memory, even the one marked as const, as they are going to be freed
+ * anyway.
+ *
+ * \param entries Pointer to the array of RzCmdDescDetails
+ */
+RZ_API void rz_cmd_desc_details_free(RzCmdDescDetail *details) {
```
Comment (XVilka) [inline librz/core/cmd/cmd_api.c:2662]: Please add `if (!details) return`

**Review by XVilka — APPROVED** [review]
The code looks ok. I am not in the favor of these conversions but it's better than the previous way of doing things anyway. Thus, I think we could proceed.

**Comment by XVilka** [comment]
Please fix python formatting:
```
Run export PATH=${HOME}/Library/Python/3.8/bin:${HOME}/Library/Python/3.9/bin:${HOME}/.local/bin:${PATH}
would reformat librz/core/cmd_descs/cmd_descs_generate.py

Oh no! 💥 💔 💥
1 file would be reformatted, 20 files would be left unchanged.
Error: Process completed with exit code 123.
```

**Pull request merged**